metomi/rose/tests/test_config.py

```diff
@@ -20,102 +20,100 @@
-        conf = metomi.rose.config.ConfigNode()
-        self.assertFalse(conf is None)
-        self.assertEqual(conf.get([]), conf)
-        self.assertFalse(conf.get(["rubbish"]))
-        node = conf.get([])
-        self.assertEqual(node.value, {})
-        node = conf.get(["rubbish"])
-        self.assertTrue(node is None)
-        self.assertTrue(conf.unset(["rubbish"]) is None)
-
-    def test_set(self):
-        """Test setting/unsetting value/ignored flag in a Config object."""
-        conf = metomi.rose.config.ConfigNode()
-        self.assertFalse(conf is None)
-        self.assertEqual(conf.set([], {}), conf)
-        conf.set(["", "top-option"], "rubbish")
-        node = conf.get(["", "top-option"])
-        self.assertEqual((node.value, node.state), ("rubbish", ""))
-        node = conf.get(["top-option"])
-        self.assertEqual((node.value, node.state), ("rubbish", ""))
-        conf.set(["rubbish"], {})
-        node = conf.get(["rubbish"])
-        self.assertEqual((node.value, node.state), ({}, ""))
-        conf.set(["rubbish", "item"], "value")
-        node = conf.get(["rubbish", "item"])
-        self.assertEqual((node.value, node.state), ("value", ""))
-        self.assertEqual(conf.get(["rubbish", "item"]).value, "value")
-        conf.get(["rubbish", "item"]).state = "!"
-        node = conf.get(["rubbish", "item"], no_ignore=True)
-        self.assertTrue(node is None)
-        self.assertEqual(conf.get(["rubbish", "item"]).value, "value")
-        conf.get(["rubbish", "item"]).state = ""
-        self.assertTrue(conf.get(["rubbish", "item"]) is not None)
-        self.assertEqual(conf.get(["rubbish", "item"]).value, "value")
-        node = conf.unset(["rubbish", "item"])
-        self.assertEqual((node.value, node.state), ("value", ""))
-        self.assertEqual(conf.unset(["rubbish", "item"]), None)
-        conf.set(["rubbish", "item"], "value", "!!")
-        node = conf.get(["rubbish", "item"])
-        self.assertEqual((node.value, node.state), ("value", "!!"))
-        self.assertTrue(conf.unset(["rubbish"]) is not None)
-        conf.set(["rubbish"], {})
-        node = conf.get(["rubbish"])
-        self.assertEqual((node.value, node.state), ({}, ""))
-        conf.set(["rubbish", "item"], "value")
-        node = conf.get(["rubbish", "item"])
-        self.assertEqual((node.value, node.state), ("value", ""))
-        conf.get(["rubbish"]).state = "!"
-        self.assertTrue(conf.get(["rubbish", "item"], True) is None)
-
-    def test_iter(self):
-        """Test the iterator"""
-        conf = metomi.rose.config.ConfigNode()
-        conf.set(["", "food"], "glorious")
-        conf.set(["dinner", "starter"], "soup")
-        conf.set(["dinner", "dessert"], "custard")
-        self.assertEqual(list(iter(conf)), ["food", "dinner"])
-        end_node = conf.get(["", "food"])
-        self.assertEqual(list(iter(end_node)), [])
-
-
-class TestConfigDump(unittest.TestCase):
+
+
+def test_init():
```

Comment:
Might be nice to have a more informative name - `test_rose_config_init`, perhaps?